meta-mender-core/classes/mender-part-images.bbclass

```diff
@@ -249,8 +249,12 @@ _MENDER_PART_IMAGE_DEPENDS = " \
     mtools-native:do_populate_sysroot \
 "
 
-_MENDER_PART_IMAGE_DEPENDS_append_mender-uboot = " u-boot:do_deploy"
-_MENDER_PART_IMAGE_DEPENDS_append_mender-grub = " grub-efi:do_deploy"
+# This assumes that U-boot is used on ARM, this could become problematic
+# if we add support other bootloaders on ARM, e.g Barebox.
+_MENDER_PART_IMAGE_DEPENDS_append_arm =     " u-boot:do_deploy"
+_MENDER_PART_IMAGE_DEPENDS_append_aarch64 = " u-boot:do_deploy"
+
+_MENDER_PART_IMAGE_DEPENDS_append_mender-grub = " grub-efi:do_deploy "
```

Comment:
We should keep the `_mender-uboot` variant in addition to the `_arm` and `_aarch64` variants, because it is required for U-Boot on x86. I don't know if anyone is using it, but I can see some potential use cases, such as strict non-GPL3 requirements.

Reply:
If I update to keep the `_mender-uboot` variant, would that be an acceptable quick fix?

I have not really looked in to what the effort would be to add a new feature.